### PR TITLE
Develop one to one documentation

### PR DESCRIPTION
**Pull request by vincentpierre**

### CODE REVIEW

File: docs/Background-TensorFlow.md [hunk 1]
```diff
@@ -17,7 +17,7 @@ performing computations using data flow graphs, the underlying representation of
 deep learning models. It facilitates training and inference on CPUs and GPUs in
 a desktop, server, or mobile device. Within the ML-Agents toolkit, when you
 train the behavior of an agent, the output is a TensorFlow model (.nn) file
-that you can then embed within a Learning Brain. Unless you implement a new
+that you can then embed within an Agent. Unless you implement a new
```
Comment (chriselion) [inline docs/Background-TensorFlow.md]: nit: "embed" -> "associate"? (I know you didn't write it)

File: docs/Basic-Guide.md [hunk 2]
```diff
@@ -216,22 +207,22 @@ INFO:mlagents.trainers: first-run-0: 3DBallLearning: Step: 10000. Mean Reward: 2
 ### After training
 
 You can press Ctrl+C to stop the training, and your trained model will be at
-`models/<run-identifier>/<brain_name>.nn` where
-`<brain_name>` is the name of the Brain corresponding to the model.
+`models/<run-identifier>/<behavior_name>.nn` where
+`<behavior_name>` is the name of the `Behavior Name` of the agents corresponding to the model.
```
Comment (chriselion) [inline docs/Basic-Guide.md]: "agent's"

File: docs/Getting-Started-with-Balance-Ball.md [hunk 3]
```diff
@@ -272,11 +252,11 @@ From TensorBoard, you will see the summary statistics:
 
 ![Example TensorBoard Run](images/mlagents-TensorBoard.png)
 
-## Embedding the Trained Brain into the Unity Environment (Experimental)
+## Embedding the Model into the Unity Environment
 
 Once the training process completes, and the training process saves the model
 (denoted by the `Saved Model` message) you can add it to the Unity project and
-use it with Agents having a **Learning Brain**.
+use it with compatible Agents (The Agents that generated the model).
```
Comment (chriselion) [inline docs/Getting-Started-with-Balance-Ball.md]: "The" -> "the"

File: docs/Basic-Guide.md [hunk 4]
```diff
@@ -73,22 +67,19 @@ if you want to [use an executable](Learning-Environment-Executable.md) or to
...
-
-   **Note:** Assigning a Brain to an agent (dragging a Brain into the `Brain` property of 
-
-the agent) means that the Brain will be making decision for that agent. If the Agent uses a
-LearningBrain either Python controls the Brain or the model on the Brain does.
+In order to setup the Agents for Training, you will need to edit the 
+`Behavior Name` under `BehaviorParamters` in the Agent Inspector window.
+The `Behavior Name` is used to group agents per behaviors. Note that Agents
+sharing the same `Behavior Name` must be agents of the same type using the
+same `Behavior Parameters` you can make sure all your agents have the same
```
Comment (surfnerd) [inline docs/Basic-Guide.md]:
```
must be agents of the same type using the
 same `Behavior Parameters` you can...
``` 
to
```
must be agents of the same type using the
 same `Behavior Parameters`.  You can...
```

File: docs/Basic-Guide.md (hunk 2 above)
Comment (surfnerd) [reply docs/Basic-Guide.md]: I think 'agents' is meant to be plural plural here

File: docs/Glossary.md [hunk 5]
```diff
@@ -6,13 +6,13 @@
   environment.
 * **Agent** - Unity Component which produces observations and takes actions in
   the environment. Agents actions are determined by decisions produced by a
-  linked Brain.
-* **Brain** - Unity Asset which makes decisions for the agents linked to it.
-* **Decision** - The specification produced by a Brain for an action to be
+  Policy.
+* **Policy** - The decision making mechanism, typically a neural network model.
```
Comment (surfnerd) [inline docs/Glossary.md]: Behavior is used a lot as @chriselion pointed out already.  Is there a reason to have both policy and behavior?

File: docs/Migrating.md [hunk 6]
```diff
@@ -6,6 +6,7 @@
 * The definition of the gRPC service has changed.
 * The online BC training feature has been removed. 
 * The BroadcastHub has been deprecated. If there is a training Python process, all LearningBrains in the scene will automatically be trained. If there is no Python process, inference will be used.
+* The Brain ScriptableObjects have been deprecated. The Brain Parameters are now on the Agent.
```
Comment (chriselion) [inline docs/Migrating.md]: "and are now referred to as BehaviorParameters"

File: docs/Basic-Guide.md (hunk 2 above)
Comment (chriselion) [reply docs/Basic-Guide.md]: oops, you're right.

File: docs/Glossary.md (hunk 5 above)
Comment (vincentpierre) [reply docs/Glossary.md]: Behavior is how I named the Agent's Parameters. Policy is the object that makes the decision and will eventually exist without Agents. So I wanted to separate the two to make it easier later to change the documentation.

**Pull request merged**